### PR TITLE
Add --password flag for providing API password

### PR DESCRIPTION
**Pull request by dotlambda**
I was not sure if it was right thing to add `HASSIO_PASSWORD` .

Should I add tests for this functionality?

### CODE REVIEW

**Comment by maxandersen** [comment]
I only didn't add it since I was always using token - hoping someone else would do it if we needed it :)

If you fix the error/issues the hound found I'm fine merging this in, having a basic test for it will be awesome. You can look at extending test_defaults.py or add your own for it.

File: homeassistant_cli/helper.py [hunk 1]
```diff
@@ -83,6 +83,7 @@ def req_raw(
     url = '{}/api/{}'.format(ctx.server, endpoint)
     headers = {
         'Authorization': 'Bearer {}'.format(ctx.token),
+        'x-ha-access': ctx.password,
```
Comment (maxandersen) [inline homeassistant_cli/helper.py]: this fix is fine but in a few hours I'll hopefully do a merge that removes this "deprecated" req_raw method so don't have to deal with passwords in two places.

File: homeassistant_cli/remote.py [hunk 2]
```diff
@@ -64,6 +64,8 @@ def restapi(
 
     if ctx.token:
         headers["Authorization"] = "Bearer {}".format(ctx.token)
+    if ctx.password is not None:
```
Comment (maxandersen) [inline homeassistant_cli/remote.py]: `if ctx.password:` should be enough.

File: homeassistant_cli/remote.py [hunk 3]
```diff
@@ -64,6 +64,8 @@ def restapi(
...
+        headers["x-ha-access"] = ctx.password
```
Comment (maxandersen) [inline homeassistant_cli/remote.py]: adding both headers will never make sense - but it also don't really hurt so let's just keep it simple.

File: homeassistant_cli/cli.py [hunk 4]
```diff
@@ -106,6 +106,10 @@ def _default_token() -> Optional[str]:
     return os.environ.get('HASS_TOKEN', os.environ.get('HASSIO_TOKEN', None))
 
 
+def _default_password() -> Optional[str]:
+    return os.environ.get('HASS_PASSWORD', os.environ.get('HASSIO_PASSWORD', None))
```
[the remark this answers is not in the document]
Comment (maxandersen) [reply homeassistant_cli/cli.py]: don't think you need this - HASSIO_PASSWORD is not a thing. HASSIO_TOKEN exist so we know to default to http://hassio/homeassistant for that special case.

You shuold just be able to use envvar="HASS_PASSWORD" in the click-option

**Comment by dotlambda** [comment]
Added a test case for `HASS_PASSWORD` to `test_defaults.py` and removed the changes to `helper.py`.

File: homeassistant_cli/remote.py (hunk 3 above)
Comment (dotlambda) [reply homeassistant_cli/remote.py]: Is there maybe some way to tell click to output an error if both, token and password, are provided?

Comment (dotlambda) [reply homeassistant_cli/remote.py]: It seems there is none but we can easily implement our own: https://github.com/pallets/click/issues/257. Should I do that?

Comment (maxandersen) [reply homeassistant_cli/remote.py]: I think that would be great, but i'm ok if you do that as a separate pr since its mainly a nice to have.

**Comment by maxandersen** [comment]
Tested this and works! Thanks @dotlambda

**Pull request merged**